### PR TITLE
docs: document partition config flag

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

**Review by kisunji — APPROVED** [review]

File: website/content/docs/agent/options.mdx [hunk 1]
```diff
@@ -1880,6 +1892,8 @@ bind_addr = "{{ GetPrivateInterfaces | include \"network\" \"10.0.0.0/8\" | attr
 
 - `segment` <EnterpriseAlert inline /> - Equivalent to the [`-segment` command-line flag](#_segment).
 
+    Note that this cannot be used with the [`partition`](#partition) option.
+
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx]:
```suggestion
    ~> **Warning**:  The `segment` option cannot be used with the [`partition`](#partition) option.

```

File: website/content/docs/agent/options.mdx [hunk 2]
```diff
@@ -465,6 +465,8 @@ The options below are all specified on the command-line.
   for more details. By default, this is an empty string, which is the `<default>`
   network segment.
 
+    Note that this cannot be used with the [`partition`](#partition) option.
+
```
[the remark this answers is not in the document]
Comment (rboyer) [reply website/content/docs/agent/options.mdx]: Turns out it doesn't seem like i can have two `code` blocks in the same line if that line is using this warning syntax and get it to render with the warning CSS.

[the remark this answers is not in the document]
Comment (rboyer) [reply website/content/docs/agent/options.mdx]: I take that back. I can't get any warning blob to style properly specifically after this `-segment` block, even if i borrow the structure from a previous flag that did something similar :thinking:

File: website/content/docs/agent/options.mdx (hunk 1 above)
Comment (rboyer) [reply website/content/docs/agent/options.mdx]: Updated with the right syntax.

File: website/content/docs/agent/options.mdx [hunk 3]
```diff
@@ -1688,6 +1690,16 @@ bind_addr = "{{ GetPrivateInterfaces | include \"network\" \"10.0.0.0/8\" | attr
...
 
+- `partition` <EnterpriseAlert inline /> - This flag is used to set
+  the name of the admin partition the agent belongs to. An agent can only join
+  and communicate with other agents within its admin partition. Review the
+  [Admin Partitions documentation](/docs/enterprise/admin-partitions) for more
+  details. By default, this is an empty string, which is the `default` admin
+  partition. This cannot be set on a server agent.
+
+    Note that this is cannot be used with either the
+    [`segment`](#segment-2) option or [`-segment`](#_segment) flag.
```
[the remark this answers is not in the document]
Comment (rboyer) [reply website/content/docs/agent/options.mdx]: Updated with the right syntax.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/528717.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit ef3a32b4ade43a42d9a77cfee58466b890c84cb0 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit ef3a32b4ade43a42d9a77cfee58466b890c84cb0 onto `release/1.11.x` succeeded!

**Pull request merged**